internal/pkg/service/stream/components.go

```diff
@@ -35,6 +35,15 @@ func StartComponents(ctx context.Context, serviceScp dependencies.ServiceScope,
 		componentsMap[c] = true
 	}
 
+	// Common task scope
+	var taskScp commonDeps.TaskScope
+	if componentsMap[ComponentAPI] {
+		taskScp, err = commonDeps.NewTaskScope(ctx, cfg.NodeID, serviceScp)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
TaskScope is required only by the `APIScope`, so it is no more part of the `ServiceScope`.